### ISSUE TITLE
Can't select map after generating random map, game freezes

### ISSUE THREAD

**Issue by ioogithub** [potmdehex/homm3tools#45]
After generating a random map I can't return to the game and select it. Here is what I see:

1. Install HD Mod 1.22
2. Run Heroes 3 HD
3. Check RMG and select h3maped.exe in the Heroes 3 complete folder
4. Click Play, New Game, Single Scenario
5. The map editor "New Window" launches.
6. Click OK to select !Original Template and all defaults
7. Map Generation Progress is displayed, progress bar finished and window is closed.
8. Return to Heroes 3, the game is now frozen

The music is playing, the cursor is movable however it is an hourglass. The game is frozen, nothing is clickable. I have to kill the app at this point to continue.

Version: HD Mod 1.22
Game: Latest Steam HD Version

**Comment by potmdehex** [comment]
Strange, never saw or had this issue reported for HDE Mod. Try running without fullscreen enabled and/or disable Steam overlay. Other than that, software such as antivirus tend to interfere with game mods.

**Comment by ioogithub** [comment]
I have tried everything you suggested. Fullscreen not enabled, team overlay off. No antivirus on this system.

I am hopeful that it may work because the editor window does open, it allows me to select a map and then the progress bar indicates it completed the map generation. Is there any log I can look at to troubleshoot?

Do you know where the random map is saved and in which folder? I can check to see that the file is actually being written.

Other than that, what is it that tells the game that is can start? It seems like the map generator tool is not passing control back to the game. If I terminate h3maped.exe then I can gain control of the game again but there is no random map to select.

**Comment by potmdehex** [comment]
Maps folder on a 64-bit system with English localization: 
C:\Program Files (x86)\Steam\SteamApps\common\Heroes of Might & Magic III - HD Edition\data\maps\EN

You can look for the newest file by timestamp there, it should be named rm.h3m. In-game it will show up in the list of maps as "Random Map".

Perhaps you have a version of h3maped.exe that HDE Mod does not fully support. It was made to work with the English version of Heroes III Complete (although was reported to work with various other localizations/versions).

**Comment by ioogithub** [comment]
Okay so I found the rm.h3m file, it was in the Heroes of Might & Magic III - HD Edition/HoMM3Complete/Maps/ folder. Is the idea that it creates the map in the HoMM3 Complete maps folder than copies it over to the HD Edition maps folder?

It looks like this is the step that isn't working. I am able to kill the game, copy the map manually then restart the game and play the map. Any idea on what is happening?

Also, everything else seems to be working- the added shortcuts and all the other features.

Thanks for the help!

**Comment by potmdehex** [comment]
Yes, it's supposed to automatically be put in the HD Edition maps folder. 

Your version of h3maped.exe may be different than the one supported by HDE Mod, although I cannot be sure. You could try using another version of Heroes III Complete (which may have the expected h3maped.exe). Another user reported problems with spaces in the path to Heroes III Complete, if you have any you could also try removing them so that Heroes III Complete is located in a path without any spaces (I never saw this problem in the past). 

Other than that, if you ever reinstall Windows or are on another machine you could try just running again as before - there are a myriad of unexpected reasons game mods sometimes don't work on certain systems (unexpected third party apps/other configurations etc).

If it still doesn't work after trying everything, you may have to just copy the map manually. Sorry about that, never had your problem reported before while this project was still actively maintained, which it hasn't been for 5 years now. Note that you don't have to restart the game after adding a map to the maps folder, simply going in and out of the maps list refreshes it.

If you do come up with a solution, it would be cool if you shared it here in case anyone else runs into your same problem. Thanks.

**Comment by luka177** [comment]
Hi! Same happens here, with mod 1.22,
